Sources/SwiftSyntaxBuilder/IntegerLiteralExprConvenienceInitializers.swift

```diff
@@ -0,0 +1,25 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+extension IntegerLiteralExpr {
+  public init(_ value: Int) {
```

Comment:
One thing, I think might be confusing is that 

`0x10` converts to the integer literal `16`. Or `1_000` represents the integer literal `1000` (without the underscore). Similarly for the float literals. I think the value of the concise syntax outweighs the cost. @akyrtzi Do you have a strong opinion on this?

Reply:
@ahoppen I'm not sure that I understand the "problem" and confusing part here?
Can you try to explain what you mean with `I think the value of the concise syntax outweighs the cost.`

Reply:
I just think that the following behavior might be confusing
```swift
func printIntegerLiteral(_ literal: IntegerLiteralExpr) {
  var text = ""
  literal.write(to: &text)
  print(text)
}

printIntegerLiteral(0x10) // prints 16, not 0x10
printIntegerLiteral(1_000) // prints 1000, not 1_000
```

But then, I think, in >95% of all cases you wouldn’t write code with hex literals or underscores and the convenience initializer does exactly what you would expect.

Reply:
I see and understand now. 

I'm currently looking into generate convenience initializers for all Expr.
So there will also be a

```swift
extension IntegerLiteralExpr {
  public init(
    digits: String
  ) {
    self.init(
      digits: Tokens.integerLiteral(digits)
    )
  }
}
```

Reply:
A string initializer also sounds good 👍 Since @akyrtzi is also on board, I think we should also continue to offer the literal initializers. Feel free to do the changes in one PR or split them up, whatever you like best.

Reply:
I will split the PRs up!